Tests/YCarouselTests/CarouselViewControllerTests.swift

```diff
@@ -82,6 +82,128 @@ final class CarouselViewControllerTests: XCTestCase {
+        sut.rightArrowKeyPressed()
+
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+    }
+
+    func test_press_keyboard_right_atindex_non_zero() {
+        let sut = makeSUT(withViews: [UIView(), UIView(), UIView()])
+        XCTAssertEqual(sut.carouselView.pages.count, 2)
+
+        // right key pressed more than numberOfPages
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+    }
+
+    func test_keyboard_disable() {
+        let sut = makeSUT(withViews: [UIView(), UIView(), UIView()])
+        XCTAssertEqual(sut.carouselView.pages.count, 2)
+
+        // before disable
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+
+        sut.isKeyboardNavigationEnabled = false
+        // after disable, no change in current page
+        sut.leftArrowKeyPressed()
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+        sut.rightArrowKeyPressed()
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+    }
+
+    func test_keycommand_left_with_vc_at_zero_index() {
+        let sut = makeSUT(withViewControllers: [
+            UIViewController(),
+            UIViewController(),
+            UIViewController()
+        ])
+        XCTAssertEqual(sut.carouselView.pages.count, 2)
+
+        // left key pressed more than once, at index 0
+        sut.leftArrowKeyPressed()
+        sut.leftArrowKeyPressed()
+        sut.leftArrowKeyPressed()
+
+        XCTAssertEqual(sut.carouselView.currentPage, 0)
+    }
+    
+    func test_keycommand_right_with_vc_at_zero_index() {
+        let sut = makeSUT(withViewControllers: [
+            UIViewController(),
+            UIViewController(),
+            UIViewController()
+        ])
+        XCTAssertEqual(sut.carouselView.pages.count, 2)
+
+        // right key pressed more than number of pages, at index 0
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+        XCTAssertEqual(sut.carouselView.currentPage, sut.carouselView.pages.count-1)
+    }
+
+    func test_keycommand_disable_with_vc() {
+        let sut = makeSUT(withViewControllers: [
+            UIViewController(),
+            UIViewController(),
+            UIViewController()
+        ])
+        XCTAssertEqual(sut.carouselView.pages.count, 2)
+
+        // before disable
+        sut.rightArrowKeyPressed()
+        sut.rightArrowKeyPressed()
+
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+
+        sut.isKeyboardNavigationEnabled = false
+        // after disable, no change in current page
+        sut.leftArrowKeyPressed()
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+        sut.rightArrowKeyPressed()
+        XCTAssertEqual(sut.carouselView.currentPage, 2)
+    }
```

Comment:
```suggestion
```
Remove

Reply:
We also don't need to assert things covered in other tests. e.g. creating a sut with 2 views for the keyboard tests we don't need to assert that the carousel has exactly 2 pages. We know from our other tests that that is the case. We can focus on the behavior for the current test (whether pressing keys does or does not cause the page to change).